include/eve/module/real/core/function/regular/generic/load.hpp

```diff
@@ -44,149 +46,57 @@ namespace eve::detail
   {
-    using type = wide<kumi::tuple<std::remove_cvref_t<decltype(*std::declval<P>())>...>,Cardinal>;
-    return perform_load(ptr, as_<type>{}, Cardinal{}, u, c);
+    return static_cast<T>(*p);
   }
 
   //================================================================================================
-  // SIMD
+  // Load from pointer - Conditional load
   //================================================================================================
-  template<typename Ptr>
-  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), Ptr ptr) noexcept
-  requires( simd_compatible_ptr<Ptr, as_wide_t<std::remove_cvref_t<decltype(*ptr)>> > )
-  {
-    return as_wide_t<std::remove_cvref_t<decltype(*ptr)>>(ptr);
-  }
-
-  template<typename Ptr, typename Cardinal>
-  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), Ptr ptr, Cardinal const &) noexcept
-  requires( simd_compatible_ptr < Ptr
-                                , as_wide_t < std::remove_cvref_t<decltype(*ptr)>
-                                            , typename Cardinal::type
-                                            >
-                                >
-          )
-  {
-    return as_wide_t<std::remove_cvref_t<decltype(*ptr)>, typename Cardinal::type>(ptr);
-  }
-
-  template<relative_conditional_expr C, typename Ptr>
-  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), C const &cond, Ptr ptr) noexcept
-  requires( simd_compatible_ptr<Ptr, as_wide_t<std::remove_cvref_t<decltype(*ptr)>> > )
-  {
-    using T = std::remove_cvref_t<decltype(*ptr)>;
-    return eve::load[cond](ptr, expected_cardinal_t<T>{});
-  }
-
-  template<relative_conditional_expr C, typename Ptr, typename Cardinal>
-  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), C const &cond, Ptr ptr, Cardinal const&) noexcept
-  requires( simd_compatible_ptr<Ptr, as_wide_t< std::remove_cvref_t<decltype(*ptr)>
-                                              , typename Cardinal::type>
-                                              >
-          )
+  template<relative_conditional_expr C, data_source Ptr, typename Pack>
+  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_)
+                            , C const &cond, safe_type const&
+                            , eve::as_<Pack>, Ptr ptr
+                            ) noexcept
+  requires simd_compatible_ptr<Ptr,Pack>
   {
-    using e_t = std::remove_cvref_t<decltype(*ptr)>;
-    using r_t = as_wide_t< e_t, typename Cardinal::type >;
+    using e_t = typename pointer_traits<Pack>::value_type;
+    using r_t = Pack;
+    using c_t = cardinal_t<Pack>;
 
     if constexpr( !std::is_pointer_v<Ptr> )
     {
-      static constexpr bool is_aligned_enough = Cardinal() * sizeof(e_t) >= Ptr::alignment();
+      static constexpr bool is_aligned_enough = c_t() * sizeof(e_t) >= Ptr::alignment();
 
       if constexpr (!sanitizers_are_on && is_aligned_enough)
       {
-        auto that = eve::unsafe(eve::load)(ptr, Cardinal{});
+        auto that = eve::unsafe(eve::load)(ptr, c_t{});
```

Comment:
`as` is lost

Reply:
Added laod(ptr,as) + variants and changed this